### PR TITLE
Cp4d 4.0

### PR DESCRIPTION
**Pull request by jeremymenke**

### CODE REVIEW

File: examples/cp4d_4.0/README.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+```hcl
+resource_group_name = "Default"
+
+// ROKS cluster parameters:
+cluster_id          = "******************"
+region                = "us-south"
+on_vpc              = false
+
+// Prereqs
+worker_node_flavor = "bc3.16x64"
```
Comment (omidmeh) [inline examples/cp4d_4.0/README.md:23]:
```suggestion
worker_node_flavor = "b3c.16x64"
```

File: examples/cp4d_4.0/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+install_watson_open_scale        = false
+install_data_virtualization      = false
+install_streams                  = false
+install_analytics_dashboard      = false
+install_spark                    = false
+install_db2_warehouse            = false
+install_db2_data_gate            = false
+install_big_sql                  = false
+install_rstudio                  = false
+install_db2_data_management      = false
```
Comment (omidmeh) [inline examples/cp4d_4.0/README.md:45]: Either these should be like the names in `variables.tf` (so become `install_wkc` for example.) Or the stuff in `main.tf` (for example at `./main.tf:75` should be updated so it reads from these new and more descriptive names.

**Review by omidmeh — CHANGES_REQUESTED** [review]

**Comment by omidmeh** [comment]
While the bedrock installs file, the Watson Studio (`wsl`) does not and the relevant pods fail saying `ImagePullBackOff`

File: examples/cp4d_4.0/README.md (hunk 2 above)
Comment (jeremymenke) [reply examples/cp4d_4.0/README.md:45]:
```suggestion
  install_wsl         = false
  install_aiopenscale = false
  install_wml         = false
  install_wkc         = false
  install_dv          = false
  install_spss        = false
  install_cde         = false
  install_spark       = false
  install_dods        = false
  install_ca          = false
  install_ds          = false
  install_db2oltp     = false
  install_db2wh       = false
  install_big_sql     = false
  install_wsruntime   = false
```

File: examples/cp4d_4.0/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+- `install_watson_open_scale`: Install Watson Open Scale module. By default it's not installed. 
+- `install_data_virtualization`: Install Data Virtualization module. By default it's not installed.
+- `install_streams`: Install Streams module. By default it's not installed.
+- `install_analytics_dashboard`: Install Analytics Dashboard module. By default it's not installed.
+- `install_spark`: Install Analytics Engine powered by Apache Spark module. By default it's not installed.
+- `install_db2_warehouse`: Install DB2 Warehouse module. By default it's not installed.
+- `install_db2_data_gate`: Install DB2 Data_Gate module. By default it's not installed.
+- `install_big_sql`: Install Big SQL module. By default it's not installed.
+- `install_rstudio`: Install RStudio module. By default it's not installed.
+- `install_db2_data_management`: Install DB2 Data Management module. By default it's not installed.
```
Comment (jeremymenke) [inline examples/cp4d_4.0/README.md:69]:
```suggestion
- `install_wsl`:  Install Watson Studio module. By default it's not installed. 
- `install_aiopenscale`: Install  Watson AI OpenScale module. By default it's not installed. 
- `install_wml`: Install Watson Machine Learning module. By default it's not installed.
- `install_wkc`: Install Watson Knowledge Catalog module. By default it's not installed.
- `install_dv`: Install Data Virtualization module. By default it's not installed.
- `install_spss`: Install SPSS Modeler module. By default it's not installed. 
- `install_cde`: Install Cognos Dashboard Engine module. By default it's not installed.  
- `install_spark`: Install Analytics Engine powered by Apache Spark module. By default it's not installed.
- `install_dods`: Install Decision Optimization module. By default it's not installed. 
- `install_ca`: Install Cognos Analytics module. By default it's not installed. 
- `install_ds`: Install DataStage module. By default it's not installed.
- `install_db2oltp`: Install Db2oltp module. By default it's not installed.
- `install_db2wh`: Install Db2 Warehouse module. By default it's not installed.         
- `install_big_sql`: Install Db2 Big SQL module. By default it's not installed.
- `install_wsruntime`: Install Jupyter Python 3.7 Runtime Addon. By default it's not installed.                                
```

Comment (jeremymenke) [inline examples/cp4d_4.0/README.md:69]:
```suggestion
- `install_wsl`:  Install Watson Studio module. By default it's not installed. 
- `install_aiopenscale`: Install  Watson AI OpenScale module. By default it's not installed. 
- `install_wml`: Install Watson Machine Learning module. By default it's not installed.
- `install_wkc`: Install Watson Knowledge Catalog module. By default it's not installed.
- `install_dv`: Install Data Virtualization module. By default it's not installed.
- `install_spss`: Install SPSS Modeler module. By default it's not installed. 
- `install_cde`: Install Cognos Dashboard Engine module. By default it's not installed.  
- `install_spark`: Install Analytics Engine powered by Apache Spark module. By default it's not installed.
- `install_dods`: Install Decision Optimization module. By default it's not installed. 
- `install_ca`: Install Cognos Analytics module. By default it's not installed. 
- `install_ds`: Install DataStage module. By default it's not installed.
- `install_db2oltp`: Install Db2oltp module. By default it's not installed.
- `install_db2wh`: Install Db2 Warehouse module. By default it's not installed.         
- `install_big_sql`: Install Db2 Big SQL module. By default it's not installed.
- `install_wsruntime`: Install Jupyter Python 3.7 Runtime Addon. By default it's not installed.                                
```

File: modules/cp4data_4.0/scripts/install-wsl.sh [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+oc project ${NAMESPACE}
+
+# ****** sed command for classic goes here *******
+if [[ ${ON_VPC} == false ]] ; then
+    sed -i -e "s/portworx-shared-gp3/ibmc-file-gold-gid/g" wsl-cr.yaml
+fi
+
+# Create wsl CR: 
+sed -i -e s#CPD_NAMESPACE#${NAMESPACE}#g wsl-cr.yaml
```
Comment (jeremymenke) [inline modules/cp4data_4.0/scripts/install-wsl.sh:34]:
```suggestion
sed -i -e "s/CPD_NAMESPACE/${NAMESPACE}/g" wsl-cr.yaml
```

File: modules/cp4data_4.0/OWNER.md [hunk 5]
```diff
@@ -0,0 +1 @@
+@cambelem
```
Comment (jeremymenke) [inline modules/cp4data_4.0/OWNER.md:1]:
```suggestion
@jeremymenke 
```

**Comment by jeremymenke** [comment]
> While the bedrock installs file, the Watson Studio (`wsl`) does not and the relevant pods fail saying `ImagePullBackOff`

Please contact us with more info as I did a fresh wsl install after seeing this comment - it worked without a fuss. Are you using "cp" for entitlement?

**Pull request merged**